check_api/src/main/java/com/google/errorprone/VisitorState.java

```diff
@@ -770,24 +771,20 @@ private static final class SharedState {
    * Like {@link Elements#getConstantExpression}, but doesn't over-escape single quotes in strings.
    */
   public String getConstantExpression(Object value) {
-    String escaped = getElements().getConstantExpression(value);
-    if (value instanceof String) {
-      // Don't escape single-quotes in string literals
-      StringBuilder sb = new StringBuilder();
-      for (int i = 0; i < escaped.length(); i++) {
-        char c = escaped.charAt(i);
-        if (c == '\\' && i + 1 < escaped.length()) {
-          char next = escaped.charAt(++i);
-          if (next != '\'') {
-            sb.append(c);
-          }
-          sb.append(next);
-        } else {
-          sb.append(c);
-        }
+    if (!(value instanceof CharSequence str)) {
+      return getElements().getConstantExpression(value);
+    }
+
+    // Don't escape single-quotes in string literals.
+    StringBuilder sb = new StringBuilder("\"");
+    for (int i = 0; i < str.length(); i++) {
+      char c = str.charAt(i);
+      if (c == '\'') {
+        sb.append('\'');
+      } else {
+        sb.append(Convert.quote(c));
```

Comment:
Incidentally I recently learned that `Convert.quote` does the thing we'd want now, but only for JDK 23+: https://github.com/openjdk/jdk/commit/144a08ee5092980859379c7664c8036eec1e264d

Reply:
Er, and as a result of that change `Convert.quote(char)` was replaced with `Convert.quote(char, boolean)`, so this change doesn't work with the latest JDK EA builds.

I'm not sure what the best option is. We could use reflection or MR-JARs to call the method.

Reply:
Ah, nice catch! I haven't looked at the MR-JAR setup yet, but indeed reflection should work. TBD whether I'll find some time today for a new pass.